backend/src/main/java/fk/prof/backend/worker/BackendDaemon.java

```diff
@@ -0,0 +1,198 @@
+        .build(vertx);
+    return httpClient;
+  }
+
+  private AggregationWindowPlannerStore buildAggregationWindowPlannerStore() {
+    return new AggregationWindowPlannerStore(
+        vertx,
+        config().getInteger("aggregation.window.duration.mins", 30),
+        config().getInteger("aggregation.window.end.tolerance.secs", 120),
+        config().getInteger("workprofile.refresh.offset.secs", 300),
+        config().getInteger("scheduling.buffer.secs", 30),
+        config().getInteger("work.assignment.max.delay.secs", 120),
+        simultaneousWorkAssignmentCounter,
+        aggregationWindowLookupStore,
+        this::getWorkFromLeader);
+  }
+
+  private void postLoadToLeader() {
+    String leaderIPAddress;
+    if((leaderIPAddress = leaderReadContext.getLeaderIPAddress()) != null) {
+
+      //TODO: load = 0.5 hard-coded right now. Replace this with dynamic load computation in future
+      float load = 0.5f;
+
+      try {
+        httpClient.requestAsync(
+            HttpMethod.POST,
+            leaderIPAddress,
+            leaderHttpPort,
+            ApiPathConstants.LEADER_POST_LOAD,
+            ProtoUtil.buildBufferFromProto(
+                BackendDTO.LoadReportRequest.newBuilder()
+                    .setIp(ipAddress)
+                    .setPort(backendHttpPort)
+                    .setLoad(load)
+                    .setCurrTick(loadTickCounter++)
+                    .build()))
+            .setHandler(ar -> {
+              if(ar.succeeded()) {
+                if(ar.result().getStatusCode() == 200) {
+                  try {
+                    Recorder.ProcessGroups assignedProcessGroups = ProtoUtil.buildProtoFromBuffer(Recorder.ProcessGroups.parser(), ar.result().getResponse());
+                    processGroupAssociationStore.updateProcessGroupAssociations(assignedProcessGroups, (processGroupDetail, processGroupAssociationResult) -> {
+                      switch (processGroupAssociationResult) {
+                        case ADDED:
+                          this.aggregationWindowPlannerStore.associateAggregationWindowPlannerIfAbsent(processGroupDetail);
+                          break;
+                        case REMOVED:
+                          this.aggregationWindowPlannerStore.deAssociateAggregationWindowPlanner(processGroupDetail.getProcessGroup());
+                          break;
+                      }
+                    });
+                  } catch (Exception ex) {
+                    logger.error("Error parsing response returned by leader when reporting load");
+                  }
+                } else {
+                  logger.error("Non OK status returned by leader when reporting load, status=" + ar.result().getStatusCode());
+                }
+              } else {
+                logger.error("Error when reporting load to leader", ar.cause());
+              }
+              setupTimerForReportingLoad();
```

Comment:
This one should be moved to finally but the ones below are in catch clause and else clause for a reason, otherwise competing load report timers will be set